samples/sample-07/src/main/java/com/example/sample07/Sample07Application.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2022-2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.sample07;
+
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+/**
+ * New consumer rebalance protocol sample which purpose is only to be used in the test assertions.
+ * See unit tests for this project for more information.
+ *
+ * @author Sanghyeok An.
+ *
+ * @since 3.3
+ */
+
+@SpringBootApplication
+public class Sample07Application {
+
+	public static void main(String[] args) {
+		SpringApplication.run(Sample07Application.class, args);
```

Comment:
Looking into this from high level, I wonder what we are trying to achieve with this sample.
Where are real consumers (`@KafkaListener` method)  for end-user purpose.
Having the test is OK, but it does not give end-users too much value since this sample really does not demonstrate (yet) how really develop an application with that new consumer group management algorithm.

Reply:
The PR in the test configuration currently has a [KafkaListener](https://github.com/spring-projects/spring-kafka/pull/3237/files#diff-5dd4447d779db9588dae1e82edfcb5e45f9cd18234a7a8851f2456cc6a02d2c5R237) where the underlying [container factory](https://github.com/spring-projects/spring-kafka/pull/3237/files#diff-5dd4447d779db9588dae1e82edfcb5e45f9cd18234a7a8851f2456cc6a02d2c5R253) uses a consumer factory where the new [consumer group protocol ](https://github.com/spring-projects/spring-kafka/pull/3237/files#diff-5dd4447d779db9588dae1e82edfcb5e45f9cd18234a7a8851f2456cc6a02d2c5R284)is used. Since this feature is in its early access form (and not recommended for production usage yet), it might still be beneficial for any users who want to play around with this in a test format. In other words, we don't expect any users to use it in a real application yet until Apache Kafka makes that recommendation, but users might be interested in trying this out via Spring Kafka, and this sample could help with that — just my 2 cents.